### PR TITLE
[DI] Improve dumped ServiceClosureArgument on PHP7+

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | 3.3
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

Replace extra closures by return types

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
AFAIK the compiled container is independent from the PHP version. This removes this property.

**Comment by chalasr** [comment]
Does it matter? With closure-proxy it was, dumping return types if available.

**Comment by nicolas-grekas** [comment]
It's a balance. What does this provide that justifies breaking the workflow of people that e.g. mix several php versions on their host (like me ;) )

**Comment by chalasr** [comment]
Fair enough, will rebase this on master and update other dumped stuff

**Comment by ogizanagi** [comment]
It's already done on master :) (https://github.com/symfony/symfony/pull/22820/commits/5772f8825cb2a8907307935401165444a1dd55f4)

File: src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php [hunk 1]
```diff
@@ -1422,14 +1422,25 @@ private function dumpValue($value, $interpolate = true)
                 if ($value instanceof ServiceClosureArgument) {
                     $value = $value->getValues()[0];
                     $code = $this->dumpValue($value, $interpolate);
+                    $return = '';
 
                     if ($value instanceof TypedReference) {
-                        $code = sprintf('$f = function (\\%s $v%s) { return $v; }; return $f(%s);', $value->getType(), ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE !== $value->getInvalidBehavior() ? ' = null' : '', $code);
+                        $nullable = ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE !== $value->getInvalidBehavior();
+                        if (70000 <= \PHP_VERSION_ID) {
+                            if (70100 > PHP_VERSION_ID || !$nullable) {
```
Comment (stloyd) [inline src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: Missing `\` to be consistent;

**Pull request closed without merging**